sources/web/datalab/auth.ts

```diff
@@ -0,0 +1,153 @@
+/*
+ * Copyright 2016 Google Inc. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+/// <reference path="../../../externs/ts/node/node.d.ts" />
+/// <reference path="../../../externs/ts/request/request.d.ts" />
+/// <reference path="../../../externs/ts/googleapis/googleapis.d.ts" />
+/// <reference path="common.d.ts" />
+
+import childProcess = require('child_process');
+import fs = require('fs');
+import google = require('googleapis');
+import http = require('http');
+import logging = require('./logging');
+
+
+var oauth2Client: any = undefined;
+
+// These are the gcloud credentials and are not actually secret.
+let clientId = '32555940559.apps.googleusercontent.com';
+let clientSecret = 'ZmssLNjJy2998hD4CTg2ejr2';
+let gcloudDir = process.env.CLOUDSDK_CONFIG || '/root/.config/gcloud';
+let userCredFile = gcloudDir + '/credentials';
+let appCredFile = process.env.GOOGLE_APPLICATION_CREDENTIALS || (gcloudDir + '/application_default_credentials.json')
```

Comment:
No, it will work. Because we set it to a location in the mapped in volume the credentials will get cached outside the container, so if you restart the container you don't have to go through auth flow again.